cljfmt/src/cljfmt/main.clj

```diff
@@ -0,0 +1,189 @@
+        incorrect (:incorrect counts 0)]
+    (when-not (zero? error)
+      (warn error "file(s) could not be parsed for formatting"))
+    (when-not (zero? incorrect)
+      (warn incorrect "file(s) formatted incorrectly"))
+    (when (and (zero? incorrect) (zero? error))
+      (println "All source files formatted correctly"))))
+
+(defn- merge-counts
+  ([]    zero-counts)
+  ([a]   a)
+  ([a b] (merge-with + a b)))
+
+(defn check
+  [options paths]
+  (let [counts (transduce
+                (comp (mapcat (partial find-files options))
+                      (map (partial check-one options))
+                      (map (fn [status]
+                             (print-file-status options status)
+                             (:counts status))))
+                (completing merge-counts)
+                paths)]
+    (print-final-count counts)
+    (exit counts)))
+
+(defn fix
+  [options paths]
+  (let [files (mapcat (partial find-files options) paths)]
+    (doseq [^java.io.File f files]
+      (let [original (slurp f)]
+        (try
+          (let [revised (reformat-string options original)]
+            (when (not= original revised)
+              (println "Reformatting" (project-path options f))
+              (spit f revised)))
+          (catch Exception e
+            (warn "Failed to format file:" (project-path options f))
+            (binding [*out* *err*]
+              (print-stack-trace e))))))))
+
+(def ^:private cli-file-reader (comp edn/read-string slurp diff/to-absolute-path))
+
+(def ^:private cli-options
+  [[nil "--file-pattern FILE_PATTERN" "The pattern that files to be formatted should match"
+    :default (:file-pattern default-config)
+    :parse-fn re-pattern]
+   [nil "--indents INDENTS_PATH" "The path to a file containing the edn data for indentation rules"
+    :parse-fn cli-file-reader]
+   [nil "--alias-map ALIAS_MAP_PATH" "The path to a file containing the edn data for the alias map"
+    :parse-fn cli-file-reader]
+   [nil "--project-root PROJECT_ROOT" "The directory to be considered the project root"
+    :default (:project-root default-config)]
+   [nil "--[no-]ansi" "Whether to colorize output"
+    :default (:ansi? default-config)
+    :id :ansi]
+   [nil "--[no-]indentation"
+    :default (:indentation? default-config)
+    :id :indentation?]
+   [nil "--[no-]remove-surrounding-whitespace"
+    :default (:remove-surrounding-whitespace? default-config)
+    :id :remove-surrounding-whitespace?]
+   [nil "--[no-]remove-trailing-whitespace"
+    :default (:remove-trailing-whitespace? default-config)
+    :id :remove-trailing-whitespace?]
+   [nil "--[no-]insert-missing-whitespace"
+    :default (:insert-missing-whitespace? default-config)
+    :id :insert-missing-whitespace?]
+   [nil "--[no-]remove-consecutive-blank-lines"
+    :default (:remove-consecutive-blank-lines? default-config)
+    :id :remove-consecutive-blank-lines?]])
+
+(defn -main
+  "Format Clojure source files"
```

Comment:
I don't think we need a docstring on the `-main` function, as it's not strictly part of the public API.